mlflow/models/evaluation/base.py

```diff
@@ -634,28 +678,38 @@ def _evaluate(
         _last_failed_evaluator = evaluator_name
         if evaluator.can_evaluate(model_type=model_type, evaluator_config=config):
             _logger.info(f"Evaluating the model with the {evaluator_name} evaluator.")
-            result = evaluator.evaluate(
+            eval_result = evaluator.evaluate(
                 model=model,
                 model_type=model_type,
                 dataset=dataset,
                 run_id=run_id,
                 evaluator_config=config,
                 custom_metrics=custom_metrics,
+                baseline_model=baseline_model,
             )
-            eval_results.append(result)
+            eval_results.append(eval_result)
 
     _last_failed_evaluator = None
 
     if len(eval_results) == 0:
-        raise ValueError(
-            "The model could not be evaluated by any of the registered evaluators, please "
-            "verify that the model type and other configs are set correctly."
+        raise MlflowException(
+            message="The model could not be evaluated by any of the registered evaluators, please "
+            "verify that the model type and other configs are set correctly.",
+            erorr_code=INVALID_PARAMETER_VALUE,
         )
 
-    merged_eval_result = EvaluationResult(dict(), dict())
+    if baseline_model:
+        merged_eval_result = EvaluationResult(dict(), dict(), dict())
+    else:
+        merged_eval_result = EvaluationResult(dict(), dict())
```

Comment:
Would it be acceptable for the constructor of `EvaluationResult()` to represent `baseline_metrics` as an empty dict by default and then populate it? That way, consumers don't have to reason about `None` vs `dict` values for the property and we can simplify this code.